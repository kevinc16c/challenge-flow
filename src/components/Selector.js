import React from 'react'
import Datos from './Datos.json'
class Selector extends React.Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this)
        this.state = {
            data: Datos
        }
    }
    
    handleChange(event){
        let value = event.target.value, localidad = {}
        const { data } = this.state
        if(value !== "0"){
            localidad = data.filter(item => item.id === value)
            this.props.onSelectLocalidad(localidad[0])
        }
    }
    render() {
        const { data } = this.state
        return (
            <select className="form-select" onChange={this.handleChange}>
                <option defaultValue value="0">Seleccione una localidad</option>
                {data.map(localidad => <option value={localidad.id}>{localidad.ciudad}</option>)}
            </select>
        )
    }
}
export default Selector