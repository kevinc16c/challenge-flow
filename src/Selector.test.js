import { render } from '@testing-library/react';
import Selector from './components/Selector';

test('renders a Selector', () => {
  const {baseElement} = render(<Selector/>);
  expect(baseElement).toMatchSnapshot();
});
