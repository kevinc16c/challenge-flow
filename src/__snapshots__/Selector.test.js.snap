// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Selector 1`] = `
<body>
  <div>
    <select
      class="form-select"
    >
      <option
        value="0"
      >
        Seleccione una localidad
      </option>
      <option
        value="3435217"
      >
        Corrientes
      </option>
      <option
        value="3836277"
      >
        Santa Fe
      </option>
      <option
        value="3860259"
      >
        Córdoba
      </option>
      <option
        value="3838233"
      >
        Salta
      </option>
      <option
        value="3843123"
      >
        Neuquén
      </option>
    </select>
  </div>
</body>
`;
